pkg/trait/deployer.go

```diff
@@ -96,6 +96,8 @@ func (t *deployerTrait) Apply(e *Environment) error {
 					} else if isIncompatibleServerError(err) {
 						t.L.Info("Fallback to client-side apply to patch resources")
 						hasServerSideApply = false
+					} else {
+						t.L.Info("Failed with server-side apply due to: %v", err.Error())
```

Comment:
That's a very good catch. In that case, what about returning the error, and let the controller log the message and requeue the request? CSA support is mainly kept for compatibility reason, and will ultimately be removed, so it should only be used as fallback when SSA is known to be unavailable, not in the general case.

Reply:
Yes, that makes sense. I just tried it earlier but found one test from container_test.go failed so I naively thought CSA might be necessary for local testing without looking into it deeply. Let me work on it with this pull req. Thanks.